WASM Microbenchmarks Not Working

Wasm Microbenchmarks are still failing after the BDN update. Two spots that pop out are a seeming failure with building, although another build of seemingly the same location succeeds after; and an inability to run a specific 'main.js' file from the browser-wasm/AppBundle folder. Respectively: 
```
// start dotnet build -c Release /p:DebugType=portable -bl:benchmarkdotnet.binlog --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true /p:NuGetPackageRoot="/home/helixbot/work/A50E08AE/p/performance/artifacts/packages" in /home/helixbot/work/A50E08AE/p/performance/artifacts/bin/MicroBenchmarks/Release/net7.0/f55d017a-085a-41d6-a734-6abcdf4c2743
[2022/02/08 12:59:15][INFO] // command took 14.99s and exited with 1
```
and 
```
[2022/02/08 13:14:47][INFO] // Execute: /home/helixbot/.jsvu/v8 --expose_wasm main.js -- --run f55d017a-085a-41d6-a734-6abcdf4c2743.dll --benchmarkName Burgers.Test2 --job "PowerPlanMode=00000000-0000-0000-0000-000000000000, Runtime=Wasm, Arguments=/p:DebugType=portable,-bl:benchmarkdotnet.binlog, Toolchain=Wasm, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1" --benchmarkId 0  in /home/helixbot/work/A50E08AE/p/performance/artifacts/bin/MicroBenchmarks/Release/net7.0/f55d017a-085a-41d6-a734-6abcdf4c2743/bin/net7.0/browser-wasm/AppBundle
[2022/02/08 13:14:47][INFO] Failed to set up high priority. Make sure you have the right permissions. Message: Permission denied
[2022/02/08 13:14:47][INFO] Error reading 'main.js'
[2022/02/08 13:14:47][INFO] No Workload Results were obtained from the run.
```

Upon further investigation, it looks like the main.js file got renamed to test-main.js. Example ls result in AppBundle:
```
dotnet.js  dotnet.timezones.blat  dotnet.wasm  icudt.dat  index.html  managed  mono-config.json  run-v8.sh  supportFiles  test-main.js
```

@naricc @SamMonoRT

Mistakenly closed this with the last issue. Although the fixes made it so the tests are running now, but we are running into some sort of timeout:
```
[2022/03/09 05:14:58][INFO] // The benchmarking process did not quit on time, it's going to get force killed now.
[2022/03/09 05:14:58][INFO] Unhandled exception. System.InvalidOperationException: Process must exit before requested information can be determined.
[2022/03/09 05:14:58][INFO]    at System.Diagnostics.Process.EnsureState(State state)
[2022/03/09 05:14:58][INFO]    at System.Diagnostics.Process.get_ExitCode()
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Toolchains.Executor.Execute(Process process, BenchmarkCase benchmarkCase, SynchronousProcessOutputLoggerWithDiagnoser loggerWithDiagnoser, ILogger logger, ConsoleExitHandler consoleExitHandler, Int32 launchIndex)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Toolchains.Executor.Execute(BenchmarkCase benchmarkCase, BenchmarkId benchmarkId, ILogger logger, ArtifactsPaths artifactsPaths, String args, IDiagnoser diagnoser, IResolver resolver, Int32 launchIndex)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Toolchains.Executor.Execute(ExecuteParameters executeParameters)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkRunnerClean.RunExecute(ILogger logger, BenchmarkCase benchmarkCase, BenchmarkId benchmarkId, IToolchain toolchain, BuildResult buildResult, IResolver resolver, IDiagnoser diagnoser, Int32 launchIndex)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkRunnerClean.Execute(ILogger logger, BenchmarkCase benchmarkCase, BenchmarkId benchmarkId, IToolchain toolchain, BuildResult buildResult, IResolver resolver)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkRunnerClean.RunCore(BenchmarkCase benchmarkCase, BenchmarkId benchmarkId, ILogger logger, IResolver resolver, BuildResult buildResult)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkRunnerClean.Run(BenchmarkRunInfo benchmarkRunInfo, Dictionary`2 buildResults, IResolver resolver, ILogger logger, List`1 artifactsToCleanup, String resultsFolderPath, String logFilePath, Int32 totalBenchmarkCount, StartedClock& runsChronometer, Int32& benchmarksToRunCount)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkRunnerClean.Run(BenchmarkRunInfo[] benchmarkRunInfos)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkSwitcher.RunWithDirtyAssemblyResolveHelper(String[] args, IConfig config, Boolean askUserForInput)
[2022/03/09 05:14:58][INFO]    at BenchmarkDotNet.Running.BenchmarkSwitcher.Run(String[] args, IConfig config)
[2022/03/09 05:14:58][INFO]    at MicroBenchmarks.Program.Main(String[] args) in /home/helixbot/work/9FAC0876/p/performance/src/benchmarks/micro/Program.cs:line 42
[2022/03/09 05:14:58][INFO] $ popd
[2022/03/09 05:14:58][ERROR] Process exited with status 134
[2022/03/09 05:14:58][INFO] Run failure registered
Traceback (most recent call last):
  File "/home/helixbot/work/9FAC0876/p/performance/scripts/benchmarks_ci.py", line 263, in <module>
    __main(sys.argv[1:])
  File "/home/helixbot/work/9FAC0876/p/performance/scripts/benchmarks_ci.py", line 241, in __main
    args
  File "/home/helixbot/work/9FAC0876/p/performance/scripts/micro_benchmarks.py", line 317, in run
    *run_args
  File "/home/helixbot/work/9FAC0876/p/performance/scripts/dotnet.py", line 471, in run
    self.working_directory)
  File "/home/helixbot/work/9FAC0876/p/performance/scripts/performance/common.py", line 227, in run
    returncode, quoted_cmdline)
subprocess.CalledProcessError: Command '$ dotnet run --project /home/helixbot/work/9FAC0876/p/performance/src/benchmarks/micro/MicroBenchmarks.csproj --configuration Release --framework net7.0 --no-restore --no-build -- --anyCategories Libraries Runtime "" --category-exclusion-filter NoInterpreter NoWASM NoMono --wasmEngine /home/helixbot/.jsvu/v8 --runtimeSrcDir /home/helixbot/work/9FAC0876/p/dotnet-wasm --aotcompilermode wasm --buildTimeout 3600 "" --partition-count 30 --partition-index 6 --artifacts /home/helixbot/work/9FAC0876/p/artifacts/BenchmarkDotNet.Artifacts --packages /home/helixbot/work/9FAC0876/p/performance/artifacts/packages --runtimes wasmnet70' returned non-zero exit status 134.
+ export _commandExitCode=1
```

--- a/eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml
+++ b/eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml
@@ -1,0 +1,15 @@
+parameters:
+  configForBuild: 'Release'
+
+steps:
+  - script: "./dotnet.sh build -p:TargetOS=Browser -p:TargetArchitecture=wasm -p:Configuration=${{ parameters.configForBuild }} /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj /t:InstallWorkloadUsingArtifacts"
+    displayName: "Install workload using artifacts"
+
+  - template: /eng/pipelines/common/upload-artifact-step.yml
+    parameters:
+      rootFolder: '$(Build.SourcesDirectory)/artifacts/'
+      includeRootFolder: true
+      displayName: Browser Wasm Artifacts
+      artifactName: BrowserWasm
+      archiveType: zip
+      archiveExtension: .zip

--- a/eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml
+++ b/eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml
@@ -2,12 +2,24 @@ parameters:
   configForBuild: 'Release'
 
 steps:
-  - script: "./dotnet.sh build -p:TargetOS=Browser -p:TargetArchitecture=wasm -p:Configuration=${{ parameters.configForBuild }} /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj /t:InstallWorkloadUsingArtifacts"
+  - script: >-
+      ./dotnet.sh build -p:TargetOS=Browser -p:TargetArchitecture=wasm /nr:false /p:TreatWarningsAsErrors=true
+      /p:Configuration=${{ parameters.configForBuild }}
+      /p:ContinuousIntegrationBuild=true
+      /t:InstallWorkloadUsingArtifacts
+      $(Build.SourcesDirectory)/src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj
     displayName: "Install workload using artifacts"
+
+  - script: >-
+      mkdir -p $(Build.SourcesDirectory)/artifacts/staging &&
+      cp -r $(Build.SourcesDirectory)/artifacts/bin/dotnet-workload $(Build.SourcesDirectory)/artifacts/staging &&
+      cp -r $(Build.SourcesDirectory)/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm $(Build.SourcesDirectory)/artifacts/staging &&
+      cp -r $(Build.SourcesDirectory)/artifacts/bin/microsoft.netcore.app.ref $(Build.SourcesDirectory)/artifacts/staging
+    displayName: "Prepare artifacts staging directory"
 
   - template: /eng/pipelines/common/upload-artifact-step.yml
     parameters:
-      rootFolder: '$(Build.SourcesDirectory)/artifacts/'
+      rootFolder: '$(Build.SourcesDirectory)/artifacts/staging'
       includeRootFolder: true
       displayName: Browser Wasm Artifacts
       artifactName: BrowserWasm

--- a/eng/pipelines/coreclr/perf.yml
+++ b/eng/pipelines/coreclr/perf.yml
@@ -57,7 +57,7 @@ jobs:
   - template: /eng/pipelines/common/platform-matrix.yml
     parameters:
       jobTemplate: /eng/pipelines/common/global-build-job.yml
-      buildConfig: release
+      buildConfig: Release
       runtimeFlavor: mono
       platforms:
       - Browser_wasm
@@ -65,20 +65,15 @@ jobs:
         buildArgs: -s mono+libs+host+packs -c $(_BuildConfig)
         nameSuffix: wasm
         isOfficialBuild: false
-        extraStepsTemplate: /eng/pipelines/common/upload-artifact-step.yml
+        extraStepsTemplate: /eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml
         extraStepsParameters:
-          rootFolder: '$(Build.SourcesDirectory)/artifacts/'
-          includeRootFolder: true
-          displayName: Browser Wasm Artifacts
-          artifactName: BrowserWasm
-          archiveType: zip
-          archiveExtension: .zip
+          configForBuild: Release
 
   #run mono wasm microbenchmarks perf job
   - template: /eng/pipelines/common/platform-matrix.yml
     parameters:
       jobTemplate: /eng/pipelines/coreclr/templates/perf-job.yml # NOTE: should we move this file out of coreclr tempelates because it contains mono jobs?
-      buildConfig: release
+      buildConfig: Release
       runtimeFlavor: wasm
       platforms:
       - Linux_x64
@@ -97,7 +92,7 @@ jobs:
   - template: /eng/pipelines/common/platform-matrix.yml
     parameters:
       jobtemplate: /eng/pipelines/coreclr/templates/perf-job.yml # note: should we move this file out of coreclr tempelates because it contains mono jobs?
-      buildconfig: release
+      buildconfig: Release
       runtimeflavor: wasm
       platforms:
       - linux_x64
@@ -131,7 +126,7 @@ jobs:
   - template: /eng/pipelines/common/platform-matrix.yml
     parameters:
       jobTemplate: /eng/pipelines/common/global-build-job.yml
-      buildConfig: release
+      buildConfig: Release
       runtimeFlavor: mono
       platforms:
       - Browser_wasm
@@ -139,14 +134,9 @@ jobs:
         buildArgs: -s mono+libs+host+packs -c $(_BuildConfig)
         nameSuffix: wasm
         isOfficialBuild: false
-        extraStepsTemplate: /eng/pipelines/common/upload-artifact-step.yml
+        extraStepsTemplate: /eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml
         extraStepsParameters:
-          rootFolder: '$(Build.SourcesDirectory)/artifacts/'
-          includeRootFolder: true
-          displayName: Browser Wasm Artifacts
-          artifactName: BrowserWasm
-          archiveType: zip
-          archiveExtension: .zip
+          configForBuild: Release
 
   # build mono for AOT
   - template: /eng/pipelines/common/platform-matrix.yml

--- a/eng/pipelines/coreclr/perf.yml
+++ b/eng/pipelines/coreclr/perf.yml
@@ -99,6 +99,7 @@ jobs:
       jobparameters:
         testgroup: perf
         livelibrariesbuildconfig: Release
+        skipLiveLibrariesDownload: true
         runtimetype: wasm
         codegentype: 'aot'
         projectfile: microbenchmarks.proj
@@ -316,6 +317,7 @@ jobs:
       jobParameters:
         testGroup: perf
         liveLibrariesBuildConfig: Release
+        skipLiveLibrariesDownload: true
         runtimeType: wasm
         codeGenType: 'wasm'
         projectFile: microbenchmarks.proj
@@ -335,6 +337,7 @@ jobs:
       jobparameters:
         testgroup: perf
         livelibrariesbuildconfig: Release
+        skipLiveLibrariesDownload: true
         runtimetype: wasm
         codegentype: 'aot'
         projectfile: microbenchmarks.proj
@@ -475,6 +478,7 @@ jobs:
       jobParameters:
         testGroup: perf
         liveLibrariesBuildConfig: Release
+        skipLiveLibrariesDownload: true
         runtimeType: wasm
         projectFile: blazor_perf.proj
         runKind: blazor_scenarios

--- a/eng/pipelines/coreclr/templates/perf-job.yml
+++ b/eng/pipelines/coreclr/templates/perf-job.yml
@@ -135,7 +135,8 @@ jobs:
           artifactName: BrowserWasm
           displayName: BrowserWasm
 
-      - script: "mkdir $(librariesDownloadDir)/bin/wasm;unzip -o $(librariesDownloadDir)/BrowserWasm/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.browser-wasm.7.0.0-ci.nupkg data/* runtimes/* -d $(librariesDownloadDir)/bin/wasm;cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/test-main.js;find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
+      - script: "mkdir $(librariesDownloadDir)/bin/wasm && unzip -o $(librariesDownloadDir)/BrowserWasm/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.browser-wasm.7.0.0-ci.nupkg data/* runtimes/* -d $(librariesDownloadDir)/bin/wasm && cp -r $(librariesDownloadDir)/BrowserWasm/artifacts/bin/dotnet-workload $(librariesDownloadDir)/bin/wasm && cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/test-main.js && find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
+      #- script: "mkdir -p $(librariesDownloadDir)/bin/wasm/data && cp -r $(librariesDownloadDir)/BrowserWasm/dotnet-workload $(librariesDownloadDir)/bin/wasm && cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/data/test-main.js && find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
         displayName: "Create wasm directory (Linux)"
 
     # Download mono AOT

--- a/eng/pipelines/coreclr/templates/perf-job.yml
+++ b/eng/pipelines/coreclr/templates/perf-job.yml
@@ -51,7 +51,7 @@ jobs:
     iosLlvmBuild: ${{ parameters.iosLlvmBuild }}
     # Test job depends on the corresponding build job
     dependsOn:
-    - ${{ if not(in(parameters.runtimeType, 'AndroidMono', 'iOSMono')) }}:
+    - ${{ if not(in(parameters.runtimeType, 'AndroidMono', 'iOSMono', 'wasm')) }}:
       - ${{ format('coreclr_{0}_product_build_{1}{2}_{3}_{4}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
     - ${{ if and(ne(parameters.liveLibrariesBuildConfig, ''), eq(parameters.skipLiveLibrariesDownload, 'false')) }}:
       - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
@@ -109,7 +109,7 @@ jobs:
           displayName: 'live-built libraries'
 
     # Download coreclr
-    - ${{ if not(in(parameters.runtimeType, 'AndroidMono', 'iOSMono')) }}:
+    - ${{ if not(in(parameters.runtimeType, 'AndroidMono', 'iOSMono', 'wasm')) }}:
       - template: /eng/pipelines/common/download-artifact-step.yml
         parameters:
           unpackFolder: $(buildProductRootFolderPath)
@@ -117,7 +117,7 @@ jobs:
           artifactName: '$(buildProductArtifactName)'
           displayName: 'Coreclr product build'
 
-    # Download mono 
+    # Download mono
     - ${{ if and(eq(parameters.runtimeType, 'mono'), ne(parameters.codeGenType, 'AOT')) }}:
       - template: /eng/pipelines/common/download-artifact-step.yml
         parameters:
@@ -206,7 +206,7 @@ jobs:
     # Create Core_Root
     - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) $(buildConfig) $(archType) generatelayoutonly $(librariesOverrideArg)
       displayName: Create Core_Root
-      condition: and(succeeded(), ne(variables.runtimeFlavorName, 'Mono'))
+      condition: and(succeeded(), ne(variables.runtimeFlavorName, 'Mono'), ne('${{ parameters.runtimeType }}', 'wasm'))
 
     # Copy the runtime directory into the testhost folder to include OOBs.
     - script: "build.cmd -subset libs.pretest -configuration release -ci -arch $(archType) -testscope innerloop /p:RuntimeArtifactsPath=$(librariesDownloadDir)\\bin\\mono\\$(osGroup).$(archType).$(buildConfigUpper) /p:RuntimeFlavor=mono;xcopy $(Build.SourcesDirectory)\\artifacts\\bin\\runtime\\$(_Framework)-$(osGroup)-$(buildConfigUpper)-$(archType)\\* $(Build.SourcesDirectory)\\artifacts\\bin\\testhost\\$(_Framework)-$(osGroup)-$(buildConfigUpper)-$(archType)\\shared\\Microsoft.NETCore.App\\7.0.0 /E /I /Y;xcopy $(Build.SourcesDirectory)\\artifacts\\bin\\testhost\\$(_Framework)-$(osGroup)-$(buildConfigUpper)-$(archType)\\* $(Build.SourcesDirectory)\\.dotnet-mono /E /I /Y;copy $(Build.SourcesDirectory)\\artifacts\\bin\\coreclr\\$(osGroup).$(archType).$(buildConfigUpper)\\corerun.exe $(Build.SourcesDirectory)\\.dotnet-mono\\shared\\Microsoft.NETCore.App\\7.0.0\\corerun.exe"

--- a/eng/pipelines/coreclr/templates/perf-job.yml
+++ b/eng/pipelines/coreclr/templates/perf-job.yml
@@ -135,8 +135,11 @@ jobs:
           artifactName: BrowserWasm
           displayName: BrowserWasm
 
-      - script: "mkdir $(librariesDownloadDir)/bin/wasm && unzip -o $(librariesDownloadDir)/BrowserWasm/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.browser-wasm.7.0.0-ci.nupkg data/* runtimes/* -d $(librariesDownloadDir)/bin/wasm && cp -r $(librariesDownloadDir)/BrowserWasm/artifacts/bin/dotnet-workload $(librariesDownloadDir)/bin/wasm && cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/test-main.js && find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
-      #- script: "mkdir -p $(librariesDownloadDir)/bin/wasm/data && cp -r $(librariesDownloadDir)/BrowserWasm/dotnet-workload $(librariesDownloadDir)/bin/wasm && cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/data/test-main.js && find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
+      - script: >-
+          mkdir -p $(librariesDownloadDir)/bin/wasm/data &&
+          cp -r $(librariesDownloadDir)/BrowserWasm/staging/dotnet-workload $(librariesDownloadDir)/bin/wasm &&
+          cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/data/test-main.js &&
+          find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \;
         displayName: "Create wasm directory (Linux)"
 
     # Download mono AOT

--- a/eng/pipelines/coreclr/templates/perf-job.yml
+++ b/eng/pipelines/coreclr/templates/perf-job.yml
@@ -19,6 +19,7 @@ parameters:
   pgoRunType: ''
   javascriptEngine: 'NoJS'
   iOSLlvmBuild: 'False'
+  skipLiveLibrariesDownload: false
 
 ### Perf job
 
@@ -52,7 +53,7 @@ jobs:
     dependsOn:
     - ${{ if not(in(parameters.runtimeType, 'AndroidMono', 'iOSMono')) }}:
       - ${{ format('coreclr_{0}_product_build_{1}{2}_{3}_{4}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
-    - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
+    - ${{ if and(ne(parameters.liveLibrariesBuildConfig, ''), eq(parameters.skipLiveLibrariesDownload, 'false')) }}:
       - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
     - ${{ if and(eq(parameters.runtimeType, 'mono'), ne(parameters.codeGenType, 'AOT')) }}:
       - ${{ format('mono_{0}_product_build_{1}{2}_{3}_{4}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
@@ -98,7 +99,7 @@ jobs:
     # Extra steps that will be passed to the performance template and run before sending the job to helix (all of which is done in the template)
 
     # Optionally download live-built libraries
-    - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
+    - ${{ if and(ne(parameters.liveLibrariesBuildConfig, ''), eq(parameters.skipLiveLibrariesDownload, 'false')) }}:
       - template: /eng/pipelines/common/download-artifact-step.yml
         parameters:
           unpackFolder: $(librariesDownloadDir)

--- a/eng/pipelines/coreclr/templates/run-scenarios-job.yml
+++ b/eng/pipelines/coreclr/templates/run-scenarios-job.yml
@@ -107,7 +107,7 @@ jobs:
       displayName: Run ci setup script (Linux)
       condition: and(succeeded(), ne(variables['Agent.Os'], 'Windows_NT'))
     # copy wasm packs if running on wasm
-    - script: cp -r $(librariesDownloadDir)/BrowserWasm/artifacts/bin/microsoft.netcore.app.runtime.browser-wasm $(PayloadDirectory);cp -r $(librariesDownloadDir)/BrowserWasm/artifacts/bin/microsoft.netcore.app.ref $(PayloadDirectory)
+    - script: cp -r $(librariesDownloadDir)/BrowserWasm/staging/microsoft.netcore.app.runtime.browser-wasm $(PayloadDirectory) && cp -r $(librariesDownloadDir)/BrowserWasm/staging/microsoft.netcore.app.ref $(PayloadDirectory)
       displayName: Copy browserwasm and runtime ref packs
       condition: and(succeeded(), eq('${{ parameters.runtimeType }}', 'wasm'))
     # copy scenario support files

--- a/eng/testing/performance/performance-setup.sh
+++ b/eng/testing/performance/performance-setup.sh
@@ -280,20 +280,9 @@ if [[ -n "$wasm_runtime_loc" ]]; then
     using_wasm=true
     wasm_dotnet_path=$payload_directory/dotnet-wasm
     mv $wasm_runtime_loc $wasm_dotnet_path
-    # install emsdk, $source_directory/src/mono/wasm/ has the nuget.config with require feed. EMSDK may be available in the payload in a different directory, should visit this install to avoid deplicated payload.
-    pushd $source_directory/src/mono/wasm/
-    make provision-wasm
-    EMSDK_PATH = $source_directory/src/mono/wasm/emsdk
-    popd
-    # wasm aot and interpreter need some source code from dotnet\runtime repo
-    rsync -aq --progress $source_directory/* $wasm_dotnet_path --exclude Payload --exclude docs --exclude src/coreclr --exclude src/tests --exclude artifacts/obj --exclude artifacts/log --exclude artifacts/tests --exclude __download__
-    # copy wasm build drop to the location that aot and interpreter build expects
-    rsync -a --progress $wasm_dotnet_path/artifacts/BrowserWasm/artifacts/* $wasm_dotnet_path/artifacts
-    rm -r $wasm_dotnet_path/artifacts/BrowserWasm/artifacts
+    extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --cli $$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/dotnet-workload/dotnet --wasmDataDir $$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/data"
     if [[ "$wasmaot" == "true" ]]; then
-        extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --runtimeSrcDir \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm --aotcompilermode wasm --buildTimeout 3600"
-    else
-        extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --runtimeSrcDir \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm"
+        extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --aotcompilermode wasm --buildTimeout 3600"
     fi
 fi
 

--- a/eng/testing/performance/performance-setup.sh
+++ b/eng/testing/performance/performance-setup.sh
@@ -280,7 +280,7 @@ if [[ -n "$wasm_runtime_loc" ]]; then
     using_wasm=true
     wasm_dotnet_path=$payload_directory/dotnet-wasm
     mv $wasm_runtime_loc $wasm_dotnet_path
-    extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --cli $$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/dotnet-workload/dotnet --wasmDataDir $$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/data"
+    extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --cli \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/dotnet-workload/dotnet --wasmDataDir \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/data"
     if [[ "$wasmaot" == "true" ]]; then
         extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --aotcompilermode wasm --buildTimeout 3600"
     fi

